src/components/interpolatedColorLegend.ts

```diff
@@ -109,6 +123,30 @@ export module Component {
       }
     }
 
+    /**
+     * Apply a gradient to the InterpolatedColorLegend.
+     *
+     * @param {boolean} isGradient Whether there should be a gradient or not.
+     *
+     * @returns {InterpolatedColorLegend} The calling InterpolatedColorLegend.
+     */
+     public gradient(isGradient: boolean): InterpolatedColorLegend {
+       this._isGradient = isGradient;
+       return this;
+     }
+
+     /**
+      * Expand the InterpolatedColorLegend to the full length/height of the chart.
+      *
+      * @param {boolean} isExpanded Whether the legend is expanded or not.
+      *
+      * @returns {InterpolatedColorLegend} The calling InterpolatedColorLegend.
+      */
+    public expand(isExpanded: boolean): InterpolatedColorLegend {
+      this._isExpanded = isExpanded;
+      return this;
+    }
```

Comment:
Ratio is also tricky since right now we essentially have a pixel length I believe of numSwatches * textHeight

I'm mainly considering the case where users would want their legend to be longer than what it currently is but not the full length of the container.